### PR TITLE
updated CI to sync with zingomobile

### PR DESCRIPTION
**Pull request by Oscar-Pepper**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -41,60 +41,134 @@ jobs:
...
+          repository: zingolabs/zingo-mobile
+      
+      - name: Check if native rust cache exists
+        id: check-build-cache
+        uses: actions/cache@v3
+        with:
+          path: android/app/src/main/jniLibs/${{ matrix.abi }}
+          key: native-rust-${{ matrix.abi }}-${{ env.CACHE-KEY }}
+          lookup-only: true
+          fail-on-cache-miss: true
```
Comment (AloeareV) [inline .github/workflows/ci.yaml:105]: This seems to be causing the integration test jobs to also fail due to the cache miss...is this intended?

**Comment by Oscar-Pepper** [comment]
Yes it is designed to fail on cache miss which triggers the build job ☺️

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (Oscar-Pepper) [reply .github/workflows/ci.yaml:105]: Are you sure this is failing integration tests? The cache is designed to fail when the rust has changed which triggers the build job

Comment (Oscar-Pepper) [reply .github/workflows/ci.yaml:105]: Looking at the logs it is the build that is failing. cI is working as intended but there is errors in the rust build

**Review by AloeareV — APPROVED** [review]
Awesome!

**Pull request merged**